### PR TITLE
[MAEB] Make seemlessm45 model more efficient

### PR DESCRIPTION
**Pull request by AdnanElAssadi56**
If you add a model or a dataset, please add the corresponding checklist:

* [dataset checklist](https://embeddings-benchmark.github.io/mteb/contributing/adding_a_dataset/#submit-a-pr)
* [model checklist](https://embeddings-benchmark.github.io/mteb/contributing/adding_a_model/#submitting-your-model-as-a-pr)

### CODE REVIEW

File: mteb/models/model_implementations/seamlessm4t_models.py [hunk 1]
```diff
@@ -137,7 +139,7 @@ def get_audio_embeddings(
 
                 all_embeddings.append(embeddings.cpu())
 
-        return torch.cat(all_embeddings, dim=0)
+        return torch.cat(all_embeddings, dim=0).numpy()
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/seamlessm4t_models.py:143]: I don't think we need the numpy conversion

File: mteb/models/model_implementations/seamlessm4t_models.py [hunk 2]
```diff
@@ -68,24 +68,26 @@ def get_audio_embeddings(
                         orig_freq=sr, new_freq=self.sampling_rate
                     )
                     array = resampler(array)
-                # truncate audio
-                array = array.cpu().detach().numpy()[:max_samples]
+                # truncate audio and convert to numpy for processor
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/seamlessm4t_models.py]: seems odd to convert it to numpy to convert it back to torch - is that really needed? (if so let us add a comment on why it is needed)

**Review by KennethEnevoldsen — APPROVED** [review]

File: mteb/models/model_implementations/seamlessm4t_models.py (hunk 2 above)
Comment (AdnanElAssadi) [reply mteb/models/model_implementations/seamlessm4t_models.py]: I think torch padding and numpy padding differ in higher dimensions i.e if audio is 2D. For most cases, it is not needed, but I did this to avoid an error.

Comment (KennethEnevoldsen) [reply mteb/models/model_implementations/seamlessm4t_models.py]: It seems like we are avoiding the error rather than fixing it - not sure why numpy and torch would differ here, are you sure that it is not the `squeeze` that makes it work?

**Comment by AdnanElAssadi** [comment]
@KennethEnevoldsen You are right, I've reviewed configs and processor accepts tensors. Should be good now.
